benchmarks/README.md

```diff
@@ -5,24 +5,35 @@ This allows us to measure how much off are we when compared with ninja-written c
 
 ## Run Types
 
-There are three types of runs: reference, tpp-mlir and libxsmm.
+There are two types of runs: reference and XSMM, for each type of benchmark: MLIR compiler and ninja code.
 
 ### Reference Runs
 
-These runs pure C without using TPP/XSMM, just executing the code as is, with compiler at -O3.
+These runs pure C++/MLIR without using TPP/XSMM, just executing the code as is, with compiler at -O3.
 This is the most naive and inefficient way of running the kernels and lets us know what is the baseline.
 
-### TPP-MLIR Runs
+The reference run is intended to produce output that will be compared with the optimized outputs.
+Both reference runs (C++ and MLIR) should also produce similar outputs.
+Since the MLIR run needs to be representative of existing Python models, the C++ side has to adapt to have the same ops.
 
-These take in the MLIR implementation and run TPP optimizations on them before running the kernels.
-The timings of these runs must be between the vanilla runs and the ninja runs, closer to the latter.
-We aim to be within 90% of the performance from ninja runs.
+Those runs can be very slow and shouldn't be used for benchmakrs for anything bigger than a single MLP layer.
+But they should be used for golden outputs for any models we try to benchmark.
+Larger models may need to cache the inputs/outputs to avoid a very slow reference run on every benchmark loop.
 
-### Ninja Runs
+### XSMM Runs
 
-These do same computations as the MLIR kernels, but calling libxsmm directly.
+These are the ninja optimized code, either by hand in C++ calling libxsmm directly, or by the Tensor Compiler, generating the calls automatically.
 The blocking/tiling/fusing parameters are in optimal configuration and should be the fastest runs.
-They also represent what the compiler _should_ be doing if it can get the transforms right.
+
+The C++ code represent what the compiler _should_ be doing if it can get the transforms right.
+The MLIR code should be within 95% of the performance from C++ runs.
+
+## How to Build
+
+The C++ benchmakrs are build by CMake and the binaries will be available in the build directory.
```

Comment:
The C++ benchmakrs are build by CMake and the binaries will be available in the build directory.
```suggestion
The C++ benchmarks are build by CMake and the binaries will be available in the build directory.
```